index.js

```diff
@@ -22,29 +22,86 @@ const os = require("os");
     // Install related tools.
     // Hopefully in the future we can install Dafny itself this way as well.
     // For now the zipped releases are simpler because they include Z3.
-    await installDotnetTool("dafny-reportgenerator", "1.*")
+    await installDotnetTool("dafny-reportgenerator", "1.*");
   } catch (error) {
     core.setFailed(error.message);
   }
 })();
 
 async function installDotnetTool(toolName, version) {
-  await exec.exec("dotnet", ["tool", "install", "-g", toolName, "--version", version])
+  await exec.exec("dotnet", [
+    "tool",
+    "install",
+    "-g",
+    toolName,
+    "--version",
+    version,
+  ]);
 }
 
 // Export functions for testing
 exports.dafnyURL = dafnyURL;
 exports.getDistribution = getDistribution;
 
-function dafnyURL(version, distribution) {
+async function dafnyURL(version, distribution) {
   const versionPath = version.startsWith("nightly") ? "nightly" : `v${version}`;
+  if (version == "nightly-latest") {
+    version = await latestNightlyVersion();
+  }
   const root = "https://github.com/dafny-lang/dafny/releases/download";
-  return `${root}/${versionPath}/dafny-${version == "2.3.0" ? "2.3.0.10506" : version}-x64-${distribution}.zip`;
+  return `${root}/${versionPath}/dafny-${
+    version == "2.3.0" ? "2.3.0.10506" : version
+  }-x64-${distribution}.zip`;
+}
+
+async function latestNightlyVersion() {
+  // Shamelessly copied and modified from dafny-lang/ide-vscode.
+  // Parsing the dotnet tool output is obviously not great,
+  // and we could consider using the NuGet API in the future.
+  // Alternatively if we move to installing Dafny using `dotnet tool install`
+  // we could use the `--prerelease` flag, although that assumes
+  // that all nightly builds use a fresh version number,
+  // and can't be used together with `--version` to express something like
+  // "install the latest 3.X version including prereleases".
+  const { exitCode, stdout, stderr } = await exec.getExecOutput(
+    "dotnet",
+    ["tool", "search", "Dafny", "--detail", "--prerelease"],
```

Comment:
This pulls both dafny and the report, I would suggest `--take 1`
There is some way to query,
but I don't understand how to get the specific package ID.
https://learn.microsoft.com/en-us/nuget/api/search-query-service-resource#search-for-packages

Reply:
Yup, I wouldn't want to just `--take 1` though, it feels brittle to depend on the order of results.

As per my comment I'd also prefer to use the NuGet API, will timebox attempting that.